src/Behat/Gherkin/Lexer.php

```diff
@@ -502,6 +502,7 @@ protected function scanTableRow()
     protected function scanTags()
     {
         $line = $this->getTrimmedLine();
+        $line = preg_replace('/\s+#.*$/', '', $line);
```

Comment:
Just ignoring comments feels wrong in the Lexer, but we don't have a consumeLineUpTo() - maybe that's a better approach so it'd be followed by a Comment with the same line number?